nixos/modules/services/misc/prowlarr.nix

```diff
@@ -0,0 +1,73 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.prowlarr;
+
+in
+{
+  options = {
+    services.prowlarr = {
+      enable = mkEnableOption "Prowlarr";
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/prowlarr/.config/Prowlarr";
+        description = "The directory where Prowlarr stores its data files.";
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Open ports in the firewall for the Prowlarr web interface.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "prowlarr";
+        description = "User account under which Prowlarr runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "prowlarr";
+        description = "Group under which Prowlarr runs.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0700 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.prowlarr = {
+      description = "Prowlarr";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
```

Comment:
Can we remove `dataDir`, `user` and `group` in favor of `DynamicUser`? We probably don't need a static uid/gid setup and `StateDirectory` takes care of permissions, so you could remove the tmpfiles rule as well.

```
DynamicUser = true;
User = "prowlarr";
StateDirectory = "prowlarr";
```